corehq/apps/api/resources/auth.py

```diff
@@ -41,10 +47,58 @@ def _inner(req, *args, **kwargs):
 
 
 class HQAuthenticationMixin:
+    decorator_map = {}  # should be set by subclasses
 
     def _get_auth_decorator(self, request):
         return self.decorator_map[determine_authtype_from_header(request)]
 
+    def get_identifier(self, request):
+        username = request.couch_user.username
+        if API_THROTTLE_WHITELIST.enabled(username):
+            return username
+        return f"{getattr(request, 'domain', '')}_{username}"
+
+
+class LoginAuthentication(HQAuthenticationMixin, Authentication):
+    """
+    Just checks you are able to login. Does not check against any permissions/domains, etc.
+    """
+    def __init__(self):
+        super().__init__()
+        self.decorator_map = {
+            'digest': digest_auth_no_domain,
+            'basic': basic_auth_no_domain,
+            'api_key': api_key_auth_no_domain,
+            'oauth2': oauth2_auth_no_domain,
+        }
+
+    def is_authenticated(self, request, **kwargs):
+        return self._auth_test(request, wrappers=[
+            self._get_auth_decorator(request),
+            #
+            # api_auth,
+        ], **kwargs)
+
+    def _auth_test(self, request, wrappers, **kwargs):
+        PASSED_AUTH = 'is_authenticated'
+
+        def dummy(request, **kwargs):
+            return PASSED_AUTH
+
+        wrapped_dummy = dummy
+        for wrapper in wrappers:
+            wrapped_dummy = wrapper(wrapped_dummy)
+
+        try:
+            response = wrapped_dummy(request, **kwargs)
+        except PermissionDenied:
+            response = HttpResponseForbidden()
+
+        if response == PASSED_AUTH:
+            return True
+        else:
+            return response
```

Comment:
Is the return type of this function expected to be treated like a boolean? I would guess that `bool(response) == True` in most cases, and therefore `self.is_authenticated(...)` will have a truthy return value even when the user is not authenticated, which seems confusing/bad. It seems like this should be changed to

```suggestion
        return response is PASSED_AUTH
```

Also, are there tests for this? I would be interested in seeing how it is intended to be used.

Reply:
this function was mostly copy/paste/modified from [the function of the same name in `LoginAndDomainAuthentication`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/api/resources/auth.py#L78-L99). I'm not sure how well it is tested which was why I was hesitant to change any of the implementation details. But I can look closer into these points when I get back next week. at a minimum it should be safe to change for this class if not the other one.

Reply:
thanks for flagging this. cleaned up in 9cae0a8214088d1bfe1c2cc18cf1ada1bc6ddff9 but did not touch the other implementation since it's used by all APIs and I don't think well tested.